pkg/apis/servicecatalog/types.go

```diff
@@ -439,3 +439,15 @@ const (
 const (
 	FinalizerServiceCatalog string = "kubernetes-incubator/service-catalog"
 )
+
+// Parameters is a union type for supporting different ways of representing parameters used for:
+// - Instance input parameters
+// - Binding output credentials
+type Parameters struct {
+	// Reference to a secret in the namespace.
+	// The parameters will be created as a dictionary from all keys in the secret.
+	SecretRef *v1.LocalObjectReference
+	// Reference to a secret key in the namespace.
+	// The parameters will be set from the JSON contained in this secret key
+	SecretKeyRef *v1.SecretKeySelector
```

Comment:
Yeah the `Optional` field concerned me as well. Okay, will replace with our own type then.